app/src/main/java/org/mozilla/fenix/components/IntentProcessors.kt

```diff
@@ -67,4 +70,8 @@ class IntentProcessors(
             FennecBookmarkShortcutsIntentProcessor(sessionManager, sessionUseCases.loadUrl)
         )
     }
+
+    val migrationIntentProcessor by lazy {
```

Comment:
What about going the other direction - overwrite `IntentProcessor` in `src/migration`? And, following that pattern for any other changes you need.

That will allow you to avoid any `support-migration` from trickling into `main`, I think.

Reply:
We tried this during the original test fit, but anything in main can't reference code in a variant. It can only work the opposite way.

See: https://github.com/mozilla-mobile/android-components/pull/5548#issuecomment-573510696

Reply:
👍